lib/dependency-manager-adapters/npm.js

```diff
@@ -53,16 +77,22 @@ module.exports = CoreObject.extend({
     });
   },
   cleanup: function() {
-    var adapter = this;
-    return adapter._restoreOriginalDependencies().then(function() {
+    return this._restoreOriginalDependencies().then(() => {
       debug('Remove backup package.json and node_modules');
-      return RSVP.all([rimraf(path.join(adapter.cwd, adapter.packageJSONBackupFileName)),
-        rimraf(path.join(adapter.cwd, adapter.nodeModulesBackupLocation))]);
+
+      let cleanupTasks = [rimraf(path.join(this.cwd, this.packageJSONBackupFileName)),
+        rimraf(path.join(this.cwd, this.nodeModulesBackupLocation))];
+
+      if (this.yarnLockPresent) {
+        cleanupTasks.push(rimraf(path.join(this.cwd, this.yarnLockBackupFileName)));
+      }
+
+      return RSVP.all(cleanupTasks);
     }).catch(function(e) {
       console.log('Error cleaning up npm scenario:', e);
     })
-    .then(function() {
-      return adapter._install();
+    .then(() => {
```

Comment:
I cannot recall our support matrix.  I'd suspect that to rely on node@4 we would need to bump our version to 0.3.x, but I'd prefer to land these "use yarn if possible" changes in 0.2.x.

So lets stick to only node@0.12 compatible syntax for now...